src/libmongoc/src/mongoc/mongoc-handshake-private.h

```diff
@@ -36,6 +36,9 @@ BSON_BEGIN_DECLS
 #define HANDSHAKE_DRIVER_NAME_MAX 64
 #define HANDSHAKE_DRIVER_VERSION_MAX 32
 
+#define HANDSHAKE_CMD_HELLO "hello"
```

Comment:
I do not have strong feelings either way. The `HANDSHAKE_CMD_LEGACY_HELLO` and `HANDSHAKE_RESPONSE_LEGACY_HELLO` were introduced to limit the usage of the terms themselves.

Reply:
I think it would be useful to have macros/constants defined for most all of our "magic strings" just to reduce cognitive overhead and chance of introducing typos or spelling errors that can't be caught by a compiler.

Reply:
There are several instances of string literal `"hello"` and `"helloOk"` being used throughout both implementation and test code that could be replaced with these new macros. Regardless whether or not these existing literals are substituted with these new macros, I think it may be worth introducing them regardless for symmetry with their legacy counterparts.

Reply:
Agreed that it would be good to use these consistently-- it was /very/ confusing (and still easily so). We also have a number of different ways that we "freehand" construct or piecewise construct handshakes-- an audit cataloguing them and consolidating them into straightforward functions would have made this MUCH easier: a great deal of my effort went into figuring out what was slightly wrong, and it was not easy to determine which examples were "right". I think that's a worthwhile project, but I want to avoid touching anything right now simply because this is feature is much behind schedule.

Reply:
I've created a new ticket to address this and related concerns:
https://jira.mongodb.org/browse/CDRIVER-4280